src/main/java/gregtech/api/metatileentity/MetaTileEntity.java

```diff
@@ -420,16 +422,21 @@ public boolean onRightClick(EntityPlayer playerIn, EnumHand hand, EnumFacing fac
                 MetaTileEntityUIFactory.INSTANCE.openUI(getHolder(), (EntityPlayerMP) playerIn);
             }
             return true;
-        } else if (playerIn.isSneaking() && playerIn.getHeldItemMainhand().isEmpty()) {
+        } else {
             EnumFacing hitFacing = hitResult.sideHit;
-
             CoverBehavior coverBehavior = hitFacing == null ? null : getCoverAtSide(hitFacing);
+            if (coverBehavior == null) {
+                return false;
+            }
+            EnumActionResult result = coverBehavior.onRightClick(playerIn, hand, hitResult);
```

Comment:
Should this return early if result is `SUCCESS`? Currently this scenario will lead to a return value of false. Is that also intended?